src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -205,20 +214,25 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | "[" ~> cond <~ "]"
     )
 
-  def cqBodyAtom: Parser[Atom] =
-    relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
-      case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  def atom = relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
+    case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  }
+  def modifierAtom = (opt("NOT") ~ "EXIST" | "OUTER") ~ "[" ~ rep1sep(cqBody, ",") ~ "]" ^^ { case (m ~ _ ~ b ~ _) =>
```

Comment:
Perhaps `atomModifier` parser is unused, and this is where should be changed.

Does the user have to write `NOT EXISTS` to emulate `ALL` constraints?  Can we actually expand that syntactic sugar from the parser and support writing `ALL` directly?  Also I thought we were going to use `!` symbol for negation.